File: EBookMasterWebSpa/src/app/components/models/book.ts
import { Author } from "./author";
import { BookBorrowing } from "./book-borrowing";
import { Category } from "./category";
import { PublishingHouse } from "./publishing-house";
import { Recommendation } from "./recommendation";
import { Series } from "./series";

export interface Book {
  id: number;
  title: string;
  publicationYear: Date;
  publishingHouse: PublishingHouse;
  series: Series | undefined;
  authors: Author[] | undefined;
  categories: Category[] | undefined;
  bookBorrowings: BookBorrowing[] | undefined;
  recommendations: Recommendation[] | undefined;
  borrowed: boolean;
  notAllowed: boolean;
}

File: EBookMasterWebApi/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using EBookMasterWebApi.Attributes;
using EBookMasterWebApi.Enums;

namespace EBookMasterWebApi.Models
{
	public class Book : Entity
	{
		[Required]
		[MaxLength(100)]
		public string Title { get; set; }

		[Required]
		public int PublishingHouseId { get; set; }

		[ForeignKey(nameof(PublishingHouseId))]
		public PublishingHouse PublishingHouse { get; set; }

		[Required]
		public DateTime PublicationYear { get; set; }

		public int? SeriesId { get; set; }

		[ForeignKey(nameof(SeriesId))]
		public Series? Series { get; set; }

		[Required]
		[AllowedBookStatus(BookStatus.Available, BookStatus.UnAvailable)]
		public BookStatus Status { get; set; }

		[Required]
		public bool IsPremium { get; set; }

		public ICollection<Author> Authors { get; set; } = new List<Author>();

		public ICollection<Category> Categories { get; set; } = new List<Category>();

		public ICollection<BookBorrowing> BookBorrowings { get; set; } = new List<BookBorrowing>();
		public ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();
	}
}
